flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/SpillableSubpartitionTest.java

```diff
@@ -192,21 +198,39 @@ public void testConsumeSpilledPartition() throws Exception {
 
 		Buffer read = reader.getNextBuffer();
 		assertNotNull(read);
+		assertNotSame(buffer, read);
+		assertFalse(read.isRecycled());
 		read.recycle();
+		assertTrue(read.isRecycled());
 
 		read = reader.getNextBuffer();
 		assertNotNull(read);
+		assertNotSame(buffer, read);
+		assertFalse(read.isRecycled());
 		read.recycle();
+		assertTrue(read.isRecycled());
 
 		read = reader.getNextBuffer();
 		assertNotNull(read);
+		assertNotSame(buffer, read);
+		assertFalse(read.isRecycled());
 		read.recycle();
+		assertTrue(read.isRecycled());
 
 		// End of partition
 		read = reader.getNextBuffer();
 		assertNotNull(read);
 		assertEquals(EndOfPartitionEvent.class, EventSerializer.fromBuffer(read, ClassLoader.getSystemClassLoader()).getClass());
+		assertFalse(read.isRecycled());
 		read.recycle();
+		assertTrue(read.isRecycled());
+
+		// finally check that the buffer has been freed after a successful (or failed) write
```

Comment:
Is this testing for failed write anywhere? Could this test catch previous error in double recycling?

Reply:
no - this is why I added numerous more tests now :) thanks for pointing this out